crates/test-framework/src/queries/validation/mod.rs

```
@@ -133,16 +158,291 @@ fn equivalent_schemas(expected_schema: &SchemaRef, actual_schema: &SchemaRef) ->
+            }
+        }
+
+        DataType::Timestamp(unit, _) => match unit {
+            TimeUnit::Second => {
+                let ts = array
+                    .as_any()
+                    .downcast_ref::<TimestampSecondArray>()
+                    .ok_or_else(|| anyhow!("Failed to downcast TimestampSecondArray"))?
+                    .value(index);
+                let dt = DateTime::from_timestamp(ts, 0)
+                    .ok_or_else(|| anyhow!("Invalid timestamp for seconds={}", ts))?;
+                Ok(Some(dt.format("%Y-%m-%d %H:%M:%S").to_string()))
+            }
+            TimeUnit::Millisecond => {
+                let ts = array
+                    .as_any()
+                    .downcast_ref::<TimestampMillisecondArray>()
+                    .ok_or_else(|| anyhow!("Failed to downcast TimestampMillisecondArray"))?
+                    .value(index);
+                let secs = ts / 1000;
+                let sub_ms = ts.rem_euclid(1000);
+                let sub_u32 = u32::try_from(sub_ms)?;
+                let nanos = sub_u32 * 1_000_000;
+                let dt = DateTime::from_timestamp(secs, nanos)
+                    .ok_or_else(|| anyhow!("Invalid timestamp"))?;
+                Ok(Some(dt.format("%Y-%m-%d %H:%M:%S%.3f").to_string()))
+            }
+            TimeUnit::Microsecond => {
+                downcast_and_format_ts!(
+                    array,
+                    index,
+                    TimestampMicrosecondArray,
+                    1_000_000,
+                    "%Y-%m-%d %H:%M:%S%.6f"
+                )
+            }
+            TimeUnit::Nanosecond => {
+                downcast_and_format_ts!(
+                    array,
+                    index,
+                    TimestampNanosecondArray,
+                    1_000_000_000,
+                    "%Y-%m-%d %H:%M:%S%.9f"
+                )
+            }
+        },
+
+        _ => Ok(None),
```

Why not return an error?